luigi/worker.py

```diff
@@ -65,6 +66,69 @@ def __init__(self, task, worker_id, result_queue, random_seed=False, worker_time
             worker_timeout = task.worker_timeout
         self.timeout_time = time.time() + worker_timeout if worker_timeout else None
 
+    @abc.abstractmethod
+    def run(self):
+        pass
+
+
+class ExternalTaskProcess(AbstractTaskProcess):
+
+    """ Checks whether an external task (i.e. one that hasn't implemented
+    `run()`) has completed. This allows us to re-evaluate the completion of
+    external dependencies after Luigi has been invoked. """
+
+    def run(self):
+        logger.info('[pid %s] Worker %s running   %s', os.getpid(), self.worker_id, self.task.task_id)
+
+        if self.random_seed:
+            # Need to have different random seeds if running in separate processes
+            random.seed((os.getpid(), time.time()))
+
+        status = FAILED
+        error_message = ''
+        try:
+            self.task.trigger_event(Event.START, self.task)
+            t0 = time.time()
+            status = None
+            try:
+                status = DONE if self.task.complete() else FAILED
```

Comment:
This class is pretty much the same as TaskProcess, but it checks the task's `complete` method instead of calling `run`.